packages/api-derive/src/staking/info.ts

```diff
@@ -2,30 +2,39 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import { ApiInterface$Rx } from '@polkadot/api/types';
+import { AccountId, BlockNumber, Exposure, Option, StakingLedger, ValidatorPrefs, UnlockChunk, Vector } from '@polkadot/types';
 import { DerivedStaking } from '../types';
 
-import { Observable, of } from 'rxjs';
+import BN from 'bn.js';
+import { combineLatest, Observable, of } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
-import { ApiInterface$Rx } from '@polkadot/api/types';
-import { AccountId, Exposure, Option, StakingLedger, ValidatorPrefs } from '@polkadot/types';
 
+import { bestNumber } from '../chain/bestNumber';
 import { drr } from '../util/drr';
+import { eraLength } from '../session/eraLength';
 
 function withStashController (api: ApiInterface$Rx, accountId: AccountId, controllerId: AccountId): Observable<DerivedStaking> {
   const stashId = accountId;
 
   return (
-    api.queryMulti([
-      [api.query.session.nextKeyFor, controllerId],
-      [api.query.staking.ledger, controllerId],
-      [api.query.staking.nominators, stashId],
-      [api.query.staking.stakers, stashId],
-      [api.query.staking.validators, stashId]
-    ]) as any as Observable<[Option<AccountId>, Option<StakingLedger>, [Array<AccountId>], Exposure, [ValidatorPrefs]]>
-  ).pipe(
-    map(([nextKeyFor, stakingLedger, [nominators], stakers, [validatorPrefs]]) => ({
+    combineLatest([
+      eraLength(api)(),
+      bestNumber(api)(),
+      api.queryMulti([
+        [api.query.session.nextKeyFor, controllerId],
+        [api.query.staking.ledger, controllerId],
+        [api.query.staking.nominators, stashId],
+        [api.query.staking.stakers, stashId],
+        [api.query.staking.validators, stashId]
+      ])
+    ]) as any as Observable<[BN, BlockNumber, [Option<AccountId>, Option<StakingLedger>, [Array<AccountId>], Exposure, [ValidatorPrefs]]]>
+  )
+  .pipe(
+    map(([eraLength, bestNumber, [nextKeyFor, stakingLedger, [nominators], stakers, [validatorPrefs]]]) => ({
       accountId,
       controllerId,
+      unlocking: calculateUnlocking(stakingLedger.unwrap().unlocking, eraLength, bestNumber),
```

Comment:
For safety, would do `unwrapOr(null)` and pass the `StakingLedger | null` into the calc